esp-hal/src/clock/mod.rs

```diff
@@ -109,39 +107,45 @@ pub trait Clock {
 pub enum CpuClock {
     /// 80MHz CPU clock
     #[cfg(not(esp32h2))]
-    Clock80MHz,
+    Clock80MHz  = 80,
+
     /// 96MHz CPU clock
     #[cfg(esp32h2)]
-    Clock96MHz,
+    Clock96MHz  = 96,
+
     /// 120MHz CPU clock
     #[cfg(esp32c2)]
-    Clock120MHz,
+    Clock120MHz = 120,
+
     /// 160MHz CPU clock
     #[cfg(not(any(esp32c2, esp32h2)))]
-    Clock160MHz,
+    Clock160MHz = 160,
+
     /// 240MHz CPU clock
     #[cfg(xtensa)]
-    Clock240MHz,
+    Clock240MHz = 240,
 }
 
-#[allow(dead_code)]
 impl Clock for CpuClock {
     fn frequency(&self) -> HertzU32 {
-        match self {
-            #[cfg(not(esp32h2))]
-            CpuClock::Clock80MHz => HertzU32::MHz(80),
-            #[cfg(esp32h2)]
-            CpuClock::Clock96MHz => HertzU32::MHz(96),
-            #[cfg(esp32c2)]
-            CpuClock::Clock120MHz => HertzU32::MHz(120),
-            #[cfg(not(any(esp32c2, esp32h2)))]
-            CpuClock::Clock160MHz => HertzU32::MHz(160),
-            #[cfg(xtensa)]
-            CpuClock::Clock240MHz => HertzU32::MHz(240),
-        }
+        HertzU32::MHz(*self as u32)
     }
 }
 
+/// Clock configuration options
+#[derive(Debug, Clone, Copy, Default)]
+pub enum ClockConfig {
+    /// Use the default clock settings after boot.
+    #[default]
+    BootDefaults,
+
+    /// Use the highest possible clock frequency.
+    Max,
+
+    /// Use a custom clock configuration.
+    Custom(CpuClock),
+}
+
```

Comment:
Instead of a separate `ClockConfig` enum, could this be an impl on `CpuClock` with `CpuClock::default()` and `CpuClock::max()`?

The thing that bothers me about `ClockConfig` is that the `Custom` value could be the max or default.